src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -121,7 +121,7 @@ def call_subprocess(
             # Convert HiddenText objects to the underlying str.
             reveal_command_args(cmd),
             stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
```

Comment:
I'm not sure. This bug was introduced in #7969 which was intended to not merge stdout and stderr, because some VCS command log warnings on stderr and we don't want to capture them in the command output. So I'd say we should leave stderr alone to be printed on the console for the user to see.

Actually I was wondering this week why I was seeing pip failing on git exit codes while not showing the error details. That is probably it.

Reply:
@Mikuana could you check if simply removing this `stderr=subprocess.STDOUT,` line works ?

Reply:
@sbidoul I will try that and let you know.

Reply:
@sbidoul that worked as well. I've updated this PR to remove that line instead of piping it to `stdout`.

Reply:
Unfortunately tests are red. It's because some calls (one actually, `via get_repository_root`) need to capture stderr.

Reply:
What would your recommendation be? Should we modify the tests, or should we trust the test and instead redirect the `stderr` to `stdout` as I had it previously?

Sorry if that seems like a silly question.

Reply:
Redirecting stderr to stdout won't work because that would reopen #7545 and #7968 where the vcs logs warnings which get mixed with the stdout we want to extract and parse.

If we let stderr go to the console, this will create unwanted noise on the console (which is why the tests fail), and bypass the pip logging and verbosity control mechanisms.

So what can we do? Not a silly question indeed.

I see two approach.

1/ The easy one is to use `Popen.communicate()` which has a safe (multithreaded) mechanism to capture stderr and stdout separately. There are two logging-related drawbacks to this: a) in debug mode it would not display the process output until it has terminated b) stdout and stderr could only be showed one after the other instead of the natural line order produced by the subprocess.

2/ The hard one is to reimplement a variant of communicate to both debug log and capture (a kind of `tee`)...

Reply:
So I went for approach 1/ in #9327